cdc panic when upgrade from v7.1.0 to master

### What did you do?

1. upgrade from v7.1.0 to master

### What did you expect to see?

CDC should not panic

### What did you see instead?

CDC panic

```
[root@zhangfubin cdcDebug]# kubectl --kubeconfig kubeconfig.yml -n cdc-upgrade-test-tps-1891409-1-353 logs -p upstream-ticdc-0
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x342fb87]

goroutine 609 [running]:
github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).GetTableStats(0xc00263a000, {0x5c, {0xc0024bb500, 0x12, 0x12}, {0xc0024bb518, 0x12, 0x12}})
        github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:981 +0x4c7
github.com/pingcap/tiflow/cdc/processor.(*processor).GetTableSpanStatus(0xc002760700, {0x5c, {0xc0024bb500, 0x12, 0x12}, {0xc0024bb518, 0x12, 0x12}}, 0x0)
        github.com/pingcap/tiflow/cdc/processor/processor.go:333 +0x138
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpan).getAndUpdateTableSpanState(0xc004074c00)
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:57 +0xad
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpan).handleAddTableTask(0xc004074c00, {0x7f17934bfb18, 0xc002500600})
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:188 +0x59e
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpan).poll(0x532a67?, {0x7f17934bfb18?, 0xc002500600?})
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:291 +0x3f
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpanManager).poll.func1({0x5c, {0xc0024bb500, 0x12, 0x12}, {0xc0024bb518, 0x12, 0x12}}, 0xc003c32778?)
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:316 +0x7a
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpanManager).poll.(*BtreeMap[...]).Ascend.func2()
        github.com/pingcap/tiflow/pkg/spanz/btree_map.go:110 +0x5a
github.com/google/btree.(*node[...]).iterate(0x4a61760, 0x4, {{{0x0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}}, 0x0}, ...}, ...)
        github.com/google/btree@v1.1.2/btree_generic.go:522 +0x5de
github.com/google/btree.(*BTreeG[...]).Ascend(0xc00286af4e?, 0x40?)
        github.com/google/btree@v1.1.2/btree_generic.go:779 +0x125
github.com/pingcap/tiflow/pkg/spanz.(*BtreeMap[...]).Ascend(...)
        github.com/pingcap/tiflow/pkg/spanz/btree_map.go:109
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableSpanManager).poll(0xc00356a2c0, {0x7f17934bfb18?, 0xc002500600?})
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:315 +0x127
github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*agent).Tick(0xc002165d40, {0x7f17934bfb18, 0xc002500600})
        github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/agent.go:214 +0x85
github.com/pingcap/tiflow/cdc/processor.(*processor).tick(0xc002760700, {0x49e76b0?, 0xc002500600})
        github.com/pingcap/tiflow/cdc/processor/processor.go:586 +0xbe
github.com/pingcap/tiflow/cdc/processor.(*processor).Tick(0xc002760700, {0x49e76b0, 0xc002500600})
        github.com/pingcap/tiflow/cdc/processor/processor.go:474 +0x3a5
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick(0xc001051720, {0x7f1793487518?, 0xc001f81d60}, {0x49bdcc0?, 0xc000ceda40?})
        github.com/pingcap/tiflow/cdc/processor/manager.go:141 +0x485
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc00085d550, {0x7f1793487518, 0xc001f81d60}, 0xc0010ae030, 0x2faf080, {0x3fa0a56, 0x9})
        github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x103d
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc0010d7900, {0x49e7570?, 0xc001f81d60}, {0x7f17933fb148, 0xc001051720}, {0x49bdcc0, 0xc000ceda40}, 0x2?, {0x3fa0a56, 0x9})
        github.com/pingcap/tiflow/cdc/capture/capture.go:577 +0x17d
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4()
        github.com/pingcap/tiflow/cdc/capture/capture.go:388 +0x42c
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.2.0/errgroup/errgroup.go:75 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 408
        golang.org/x/sync@v0.2.0/errgroup/errgroup.go:72 +0x96

```

### Versions of the cluster

cdc version: master

```
"[version.go:47] [\"Welcome to Change Data Capture (CDC)\"] [release-version=v7.4.0-alpha] [git-hash=dd8da8e725052e3abb8be12513f74ebb5e479a22] [git-branch=heads/refs/tags/v7.4.0-alpha] [utc-build-time=\"2023-08-12 11:36:23\"] [go-version=\"go version go1.21.0 linux/amd64\"] [failpoint-build=false]"
```

--- a/cdc/processor/sinkmanager/manager.go
+++ b/cdc/processor/sinkmanager/manager.go
@@ -33,6 +33,7 @@ import (
 	"github.com/pingcap/tiflow/cdc/sink/dmlsink/factory"
 	tablesinkmetrics "github.com/pingcap/tiflow/cdc/sink/metrics/tablesink"
 	"github.com/pingcap/tiflow/cdc/sink/tablesink"
+	"github.com/pingcap/tiflow/pkg/config"
 	cerror "github.com/pingcap/tiflow/pkg/errors"
 	"github.com/pingcap/tiflow/pkg/retry"
 	"github.com/pingcap/tiflow/pkg/spanz"
@@ -979,7 +980,12 @@ func (m *SinkManager) GetTableStats(span tablepb.Span) TableStats {
 	m.sinkMemQuota.Release(span, checkpointTs)
 	m.redoMemQuota.Release(span, checkpointTs)
 
-	stuckCheck := time.Duration(*m.changefeedInfo.Config.Sink.AdvanceTimeoutInSec) * time.Second
+	advanceTimeoutInSec := util.GetOrZero(m.changefeedInfo.Config.Sink.AdvanceTimeoutInSec)
+	if advanceTimeoutInSec <= 0 {
+		log.Warn("AdvanceTimeoutInSec is not set, use default value", zap.Any("sinkConfig", m.changefeedInfo.Config.Sink))
+		advanceTimeoutInSec = config.DefaultAdvanceTimeoutInSec
+	}
+	stuckCheck := time.Duration(advanceTimeoutInSec) * time.Second
 	if version > 0 && time.Since(advanced) > stuckCheck &&
 		oracle.GetTimeFromTS(tableSink.getUpperBoundTs()).Sub(oracle.GetTimeFromTS(checkpointTs.Ts)) > stuckCheck {
 		log.Warn("Table checkpoint is stuck too long, will restart the sink backend",

--- a/pkg/config/replica_config.go
+++ b/pkg/config/replica_config.go
@@ -68,7 +68,7 @@ var defaultReplicaConfig = &ReplicaConfig{
 		OnlyOutputUpdatedColumns:         util.AddressOf(false),
 		DeleteOnlyOutputHandleKeyColumns: util.AddressOf(false),
 		TiDBSourceID:                     1,
-		AdvanceTimeoutInSec:              util.AddressOf(uint(150)),
+		AdvanceTimeoutInSec:              util.AddressOf(DefaultAdvanceTimeoutInSec),
 	},
 	Consistent: &ConsistentConfig{
 		Level:             "none",

--- a/pkg/config/sink.go
+++ b/pkg/config/sink.go
@@ -29,6 +29,8 @@ import (
 const (
 	// DefaultMaxMessageBytes sets the default value for max-message-bytes.
 	DefaultMaxMessageBytes = 10 * 1024 * 1024 // 10M
+	// DefaultMaxMessageCount sets the default value for advance-timeout-in-sec.
+	DefaultAdvanceTimeoutInSec = uint(150)
 
 	// TxnAtomicityKey specifies the key of the transaction-atomicity in the SinkURI.
 	TxnAtomicityKey = "transaction-atomicity"
@@ -435,7 +437,7 @@ func (s *SinkConfig) validateAndAdjust(sinkURI *url.URL) error {
 		}
 	}
 
-	if s.AdvanceTimeoutInSec != nil && *s.AdvanceTimeoutInSec == 0 {
+	if util.GetOrZero(s.AdvanceTimeoutInSec) == 0 {
 		return cerror.ErrSinkInvalidConfig.GenWithStack("advance-timeout-in-sec should be greater than 0")
 	}
 

--- a/pkg/config/sink.go
+++ b/pkg/config/sink.go
@@ -29,7 +29,7 @@ import (
 const (
 	// DefaultMaxMessageBytes sets the default value for max-message-bytes.
 	DefaultMaxMessageBytes = 10 * 1024 * 1024 // 10M
-	// DefaultMaxMessageCount sets the default value for advance-timeout-in-sec.
+	// DefaultAdvanceTimeoutInSec sets the default value for advance-timeout-in-sec.
 	DefaultAdvanceTimeoutInSec = uint(150)
 
 	// TxnAtomicityKey specifies the key of the transaction-atomicity in the SinkURI.

--- a/pkg/config/sink.go
+++ b/pkg/config/sink.go
@@ -438,7 +438,8 @@ func (s *SinkConfig) validateAndAdjust(sinkURI *url.URL) error {
 	}
 
 	if util.GetOrZero(s.AdvanceTimeoutInSec) == 0 {
-		return cerror.ErrSinkInvalidConfig.GenWithStack("advance-timeout-in-sec should be greater than 0")
+		log.Warn(fmt.Sprintf("advance-timeout-in-sec is not set, use default value: %d seconds", DefaultAdvanceTimeoutInSec))
+		s.AdvanceTimeoutInSec = util.AddressOf(DefaultAdvanceTimeoutInSec)
 	}
 
 	return nil

